runtime/compiler/x/codegen/J9CodeGenerator.cpp

```diff
@@ -353,6 +353,33 @@ J9::X86::CodeGenerator::endInstructionSelection()
       }
    }
 
+TR::VectorLength
+J9::X86::CodeGenerator::getMaxPreferredVectorLength()
+   {
+   TR::CPU *cpu = &self()->comp()->target().cpu;
+
+   // TODO: More research is needed
+   //
+   // Regressive CPU frequency scaling when using 256 or 512-bit vectorization is
+   // a known issue on intel x86 hardware. Some microarchitectures are known to be
+   // more affected than others.
+   if (cpu->supportsFeature(OMR_FEATURE_X86_AVX2))
+      {
+      // A CPU frequency regression of 8% was observed on Haswell with a 256-bit
+      // vectorized hashcode intrinsic. This effect was not explicitly seen or
+      // reported during performance testing on other microarchitectures. However,
+      // this does not definitively mean the regression is absent on these newer
+      // architectures. We need to evaluate magnitude of CPU frequency scaling
+      // impacts on various microarchitectures.
```

Comment:
This kind of justification comment should go in the PR description motivating the change. I don't think it belongs in the code. This is a generally useful function on its own.

You can say the choice of vector length is empirically driven.